src/playercards/DeckBuilder.ttslua

```diff
@@ -0,0 +1,543 @@
+    broadcastToAll("Investigator Detected: " .. info.collision_object.getName(), Color.Green)
+    if investigatorData.deck_requirements.signatures then
+      local cardsToSpawn = {}
+      for cardId, cardCount in pairs(investigatorData.deck_requirements.signatures) do
+        local card = allCardsBagApi.getCardById({ id = cardId })
+        if card ~= nil then
+          for i = 1, cardCount do
+            table.insert(cardsToSpawn, { data = card.data, metadata = card.metadata})
+          end
+        end
+      end
+      Spawner.spawnCards(
+        cardsToSpawn,
+        { x = 53.22, y = 2, z = 46.46},
+        { x = 0.00, y = 270.00, z = 0.00},
+        true,
+        function() end
+      )
+    end
+    if investigatorData.deck_requirements.choices then
+      createChoiceButtons()
+    end
+  elseif info.collision_object.hasTag("DeckBuilderToken") then
+    importDeck(info.collision_object)
+  end
+end
+
+function onCollisionExit(info)
+  local objectData = info.collision_object.getData()
+  log(investigatorObject == info.collision_object)
+  if objectData.Name == "Deck" then
+    removeDeck()
+  elseif investigatorObject == info.collision_object then
+    investigatorData = nil
+    investigatorObject = nil
+    playerCardPanelApi.deleteAll("middle")
+    resetChoiceButtons()
+  end
+end
+
+function onObjectDestroy(object)
+  if object == deckObject then
+    removeDeck()
+  elseif object == investigatorObject then
+    investigatorData = nil
+    investigatorObject = nil
+    playerCardPanelApi.deleteAll("middle")
+    resetChoiceButtons()
+  end
+end
+
+function createChoiceButtons()
+  local buttonCount = 3
+
+  for _, cardSpec in ipairs(investigatorData.deck_options) do
+    if cardSpec.choiceName then
+      choiceButtonParameters.position   = choiceButtonStatus[buttonCount].position
+      choiceButtonParameters.label      = cardSpec.choiceName
+      choiceButtonParameters.click_function = "toggleChoiceButton" .. buttonCount
+      self.createButton(choiceButtonParameters)
+      choiceButtonStatus[buttonCount].spawned = true
+      choiceButtonStatus[buttonCount].label = choiceButtonParameters.label
+      buttonCount = buttonCount + 1
+    end
+  end
+
+end
+
+function resetChoiceButtons()
+  for i, button in pairs(choiceButtonStatus) do
+    if button.spawned then
+      self.removeButton(i)
+      choiceButtonStatus[i].spawned   = false
+      choiceButtonStatus[i].selected  = false
+      choiceButtonStatus[i].label     = ""
+    end
+  end
+  choiceCount = 0
+end
+
+function toggleChoiceButton3(_, _, alt_click)
+  toggleChoiceButton(3, alt_click)
+end
+
+function toggleChoiceButton4(_, _, alt_click)
+  toggleChoiceButton(4, alt_click)
+end
+
+function toggleChoiceButton5(_, _, alt_click)
+  toggleChoiceButton(5, alt_click)
+end
+
+function toggleChoiceButton6(_, _, alt_click)
+  toggleChoiceButton(6, alt_click)
+end
+
+function toggleChoiceButton7(_, _, alt_click)
+  toggleChoiceButton(7, alt_click)
+end
```

Comment:
You can append a parameter to click_functions, either via using `setVar` or `_G`. Example:
https://github.com/argonui/SCED/blob/fea12454dee7979913fb6b8d69c38d798ad14288/src/accessories/ChaosBagManager.ttslua#L41C1-L44C47